cmd/kubeadm/app/util/tokens.go

```diff
@@ -103,18 +105,45 @@ func UseGivenTokenIfValid(s *kubeadmapi.Secrets) (bool, error) {
 	return true, nil // given and valid
 }
 
-func CreateToken(client *clientset.Clientset, tokenSecret *kubeadmapi.Secrets, tokenDuration time.Duration) error {
-	secret := &api.Secret{
-		ObjectMeta: api.ObjectMeta{
-			Name: fmt.Sprintf("%s%s", BootstrapTokenSecretPrefix, tokenSecret.TokenID),
-		},
-		Type: api.SecretTypeBootstrapToken,
-		Data: encodeTokenSecretData(tokenSecret, tokenDuration),
-	}
-	if _, err := client.Secrets(api.NamespaceSystem).Create(secret); err != nil {
-		return fmt.Errorf("<util/tokens> failed to create bootstrap token [%v]", err)
+// UpdateOrCreateToken attempts to update a token with the given ID, or create if it does
+// not already exist.
+func UpdateOrCreateToken(client *clientset.Clientset, tokenSecret *kubeadmapi.Secrets, tokenDuration time.Duration) error {
+	secretName := fmt.Sprintf("%s%s", BootstrapTokenSecretPrefix, tokenSecret.TokenID)
```

Comment:
The only "hole" here is if someone creates a bootstrap token from some other client with a different name but the same ID.  The "right" way to do this would be to list all the tokens and index those by ID.  Then you could update a token without making assumptions about the secret name.

But that probably isn't worth the trouble at this point.  As things stand the server impl will throw a warning and do its best.  Folks can always use a `kubectl delete secret` command to clean this up if need be.